lib/Translation/ExportVerilog/ExportVerilog.cpp

```diff
@@ -412,6 +412,37 @@ getLocationInfoAsString(const SmallPtrSet<Operation *, 8> &ops) {
   return sstr.str();
 }
 
+/// Given string \p origName, generate a new name if it conflicts with any
+/// keyword or any other name in the set \p recordNames. Use the int \p
+/// nextGeneratedNameID as a counter for suffix. Update the \p recordNames with
+/// the generated name and return an iterator corresponding to the new entry.
+llvm::StringMapIterator<llvm::NoneType>
+resolveKeywordConflict(StringRef origName, llvm::StringSet<> &recordNames,
+                       size_t &nextGeneratedNameID) {
+  auto name = origName;
+  // Get the list of reserved words we need to avoid.  We could prepopulate this
+  // into the used words cache, but it is large and immutable, so we just query
+  // it when needed.
+  auto &reservedWords = getReservedWords();
+  SmallVector<char, 16> nameBuffer(name.begin(), name.end());
+  nameBuffer.push_back('_');
+  auto baseSize = nameBuffer.size();
+
+  // Loop until we get a name that is not a keyword and is unique.
+  while (reservedWords.count(name) || recordNames.count(name)) {
+
+    // We need to auto-unique it.
+    auto suffix = llvm::utostr(nextGeneratedNameID++);
+    nameBuffer.append(suffix.begin(), suffix.end());
+    name = StringRef(nameBuffer.data(), nameBuffer.size());
+
+    // Chop off the suffix and try again until we get a unique name..
+    nameBuffer.resize(baseSize);
+  }
+  auto insertIt = recordNames.insert(name);
+  return insertIt.first;
```

Comment:
It's a small optimization, but in the normal case of this loop is doing a "count(name)" followed by "insert(name)" which does two lookups.  You could make it more efficient by using the .second member returned by insert.  Something like this should work:

```
while (1) {
  if (!reservedWords.count(name)) {
     auto itAndInserted = recordNames.insert(name);
     if (itAndInserted.second)  // success!
       return itAndInserted.first->getKey();
     // Conflict so go unique.
  }

  name = autounique logic you already have
}
```

Reply:
Done.